jax/_src/sharding_impls.py

```diff
@@ -672,13 +703,40 @@ def __init__(self, devices: Sequence[xc.Device] | np.ndarray,
 
   @property
   def shape(self):
```

Comment:
In general, there is no need to document class properties.

Same for everything marked with `@property` or `@cached_property` below.